library/RevAPI/Rev.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+namespace RevAPI;
+
+use Guzzle\Http\Client as HttpClient;
+use Guzzle\Http\Exception\BadResponseException;
+use Guzzle\Http\Message\RequestInterface;
+
+class Rev {
+    const PRODUCTION_HOST = 'www.rev.com';
+    const SANDBOX_HOST    = 'api-sandbox.rev.com';
+    
+    protected $base_api_url;
+
+    /**
+     * @var HttpClient
+     */
+    protected $http_client;
+    
+    public function __construct($client_api_key, $user_api_key, $host = null, $http_config = array())
+    {
+        if (null == $host) {
+            $host = self::PRODUCTION_HOST;
+        }
+        
+        $http_config['request.options']['headers']['Authorization'] = 'Rev ' . $client_api_key .':' . $user_api_key;
+        $http_config['request.options']['headers']['Content-Type'] = 'application/json';
+        
+        $this->base_api_url = 'https://' . $host . '/api/v1/';
+        
+        $this->http_client = new HttpClient($this->base_api_url, $http_config);
+    }
+
+    /**
+     * Send a request and convert any BadResponseExceptions to a RequestException
+     * 
+     * @param RequestInterface $request
+     * @return \Guzzle\Http\Message\Response
+     * @throws Exception\RequestException
+     */
+    protected function sendRequest(RequestInterface $request)
+    {
+        try {
+            return $request->send();
+        } catch (BadResponseException $e) {
+            throw new Exception\RequestException($e);
+        }
+    }
+
+    /**
+     * Get all orders
+     *
+     * @param int $page - the number of the page of orders to return. Optional, by default returns the first page
+     * @param int $page_size - the number of orders per page to return. Optional, by default returns 25 orders per page. Page size has to be between 5 and 100
+     * @return Orders
+     * @throws Exception\RequestException
+     */
+    public function getOrders($page = 0, $page_size = 25)
+    {
+        $request = $this->http_client->get('orders?page=' . $page . '&page_size=' . $page_size);
```

Comment:
It might be worth doing some type checking on all these functions that embed the parameters into API request URLs. Without validating that are the proper type, a developer could theoretically inject parameters into the URL.